### DISCUSSION TITLE
Working out formulae for effects of weight on walk/run

### DISCUSSION THREAD

**Discussion by I-am-Erk** [CleverRaven/Cataclysm-DDA#58444] [Ideas]
I'm posting some preliminary equations here in response to #58380 discussion.

There are a few related issues that need solving here, some of which have been in the works for a while.

### BMR and Cardio
I used BMR in the initial [Cardio](https://github.com/CleverRaven/Cataclysm-DDA/issues/44370) proposal because I thought it would make it easier to infer other things that should affect stamina, but it has turned out to be a can of worms as it leads to larger mutants and obese characters getting extra stamina. Instead I think we should change the BMR part of this calculation to a flat number (1000) and ensure that anything that any mutations etc. that change metabolic rate straight up also change that flat cardio mod by the same amount. This should accomplish the same goal.

Note that 1000 leads to a bit higher stamina to start than currently. That's because we're about to penalize stamina with weight.

### Weight and movement stamina
Your total weight - personal weight and carried - should have an effect on your movement stamina costs, walking and running. When walking, this would then drop your overall walk speed, as your character adjusts their walk speed to ensure they aren't losing stamina. When sprinting, this doesn't change you speed but does shorten the duration you can sprint.

Note that this should only impact movement costs, not any other stamina expenditures. That allows us to try this out without huge repercussions to overall game balance.

I am still looking for sources to get a good calculation here, but I am thinking something like:

`neutral_weight = strength*10`

This is the amount of weight, in kg, that your character can carry without extra penalty. For an average character this is about 80kg, and an average character weighs around 70kg, so this means you can wear normal clothes and carry some gear without noticing any change to your stamina. I would like to work out specific numbers but I imagine after this, we do something like increase stamina costs for movement by 10% for each 10% you are over your neutral carry weight.

### Weight and movement speed
Your total weight should not impact your walking speed any more than by the stamina calculation changes, but we may want to reduce your sprinting speed by something like 1/2 of the amount we increased your stamina by. Again this will need cross-checking to academic sources. So if you use 10% extra stamina, your sprint speed would decrease by 5%.

### Athletics and movement speed
Your athletics skill should help counteract penalties to movement by letting you move faster with no additional movement cost. I suggest this be an exponential rise, but not too extreme at high levels, something like:

`movement_bonus = (athetics_skill/2)^2` as a percent

This means at 10 athletics you get a 25% boost to base movement speed when walking, *after* stamina calculations. This is further increased when sprinting, again, *after* stamina checks. This compounds with the effect of athletics on total stamina, although that is comparatively small.

I'm also open to suggestions about if athletics knowledge level should have a similar smaller effect, representing someone who has been athletic but lost some of their training.

This should go along with making movement train athletics, probably with the amount of training based on the stamina you spend, so that carrying heavy weights or sprinting has a more significant effect. This training should, ideally, not affect your focus.

### Mutations and carry weight
We may want to have an additional calculation for certain mutations (large and small size specifically) that increase or decrease the effective impact of your carried (not body) weight. I suggest that small size mutations increase the effective carry weight by 10% and large size decrease it by 10%, possibly with the 'unwieldy' mutations giving you less benefit than the stable ones. This represents the fact that it's mechanically more challenging to carry the same burdens on a smaller frame due to leverage and leg length and things, without getting into too much detail.

### Potential optimizations
We may want to cache your total weight value and only recheck it when you begin movement, open inventory or drop item, and otherwise every (5?) seconds; I imagine it will be somewhat intensive to get the weight of all your inventory stuff and we don't need to do it every tick. There are other ways to change carry weight like Throw and the more of these cases automatically catch and update the weight the less frequently we need to update the cache to catch missed times.

### NPCs
I think this would be reasonable stuff to include on NPCs as well, even though they currently don't have stamina: particularly the effect of athletics and carry weight on movement rate.

### Gaps in this so far
What this currently doesn't do a good job on is modeling the benefit of being lightly burdened. Presently, lightly burdened would be slightly buffed compared to current because your base stamina would be a bit higher. There is no strong advantage to being a 50kg person in running gear versus a 70kg person with full clothing and a hiking pack, at the same level of strength and athletics. Also this doesn't model the effect of inanition and being underweight on stamina. Those are things we should look at and possibly revise this model.

**Comment by I-am-Erk** [comment]
The main thing I am unsure of thinking about this (besides the actual formulae which I just made up whole cloth) is if we should have some difference between body weight, worn weight, and weight in things that are hanging off you (thanks to our new hanging layers and things). This would make calculations way harder to write up, and probably a little slower, but it would also allow us to make the effects more robust. For example, your worn weight and body weight should have less impact than your big swingin' pack, and should be less impacted by size. This in turn could let us differentiate between that 50kg person with a 30kg kit, compared to a 70kg person of equal size and strength with a 10kg kit. What I am not convinced of is if those *should* be differentiated. Other things being equal, does that make a significant difference?

A lot of that can also just be modeled with encumbrance. We could keep this to *just* weight, and if your load is unwieldy we can represent that with encumbrance impacting appropriate stuff.

**Comment by I-am-Erk** [comment]
Regarding inanition, ie. the wasting effect of malnutrition and starvation, I think that would best be modeled simply as an effect that drops your stamina, stamina regen, your strength, and perhaps your max carry weight as well (double dipping on strength, but that allows it to have a less profound drop on the strength stat while still having a big impact on the game)

**Comment by Night-Pryanik** [comment]
I suppose we should also think over crouch and prone movement modes. Moving in these modes should pose additional difficulties for overweight characters.

**Reply by I-am-Erk** [reply]
They should at least be affected by the same stamina and movement changes. I am not sure what other changes we'd implement, particularly not on a first pass
